module/os/freebsd/zfs/zfs_vnops_os.c

```diff
@@ -6089,6 +6089,79 @@ zfs_freebsd_aclcheck(struct vop_aclcheck_args *ap)
 	return (EOPNOTSUPP);
 }
 
+static int
+zfs_advise(vnode_t *vp, size_t start, size_t end, int advice, cred_t *cr,
+    caller_context_t *ct)
```

Comment:
cr and ct are unused, I think they should just be removed altogether.

perhaps the routine would be nicer if the vfs stuff was handled prior to calling into it

Reply:
Yeah, they were useless.
The extra function is also kinda useless.  Let's just do it in one?  New version pushed like that.